libraries/kotlin.test/common/src/test/kotlin/kotlin/test/tests/BasicAssertionsTest.kt

```diff
@@ -205,11 +205,35 @@ class BasicAssertionsTest {
     fun testExpectFails() {
         checkFailedAssertion { expect(1) { 2 } }
     }
+
+    @Test
+    fun testAssertIs() {
+        val s: Any = "test"
+        assertIs<String>(s)
+        assertEquals(4, s.length)
+    }
+
+    @Test
+    fun testAssertIsFails() {
+        val message = "Expected value to be of type <kotlin.Int>, actual <class kotlin.String>."
```

Comment:
Note that different platforms can have different `toString` representations of `KType` and `KClass`. Better to check that the message contains essential information rather than to expect it in exact form.